sql/sql_table.cc

```diff
@@ -3372,7 +3385,17 @@ bool mysql_rm_table_no_locks(THD *thd, TABLE_LIST *tables, bool if_exists,
 
       thd->thread_specific_used = true;
 
-      if (built_query.write_bin_log()) goto err_with_rollback;
+      if (thd->variables.binlog_ddl_query_log_events) {
+        if (thd->binlog_query(
+                THD::STMT_QUERY_TYPE, thd->query().str, thd->query().length,
+                drop_ctx.has_base_atomic_tables(), false /* direct */,
+                false /* suppress_use */, 0 /* errcode */)) {
+          goto err_with_rollback;
+        }
+
```

Comment:
I think abort after engine commit is not good idea. 5.7 doesn't have atomic DDL. If you want to abort, better to do it at start of DDL? (or atleast after server decides that this is multi table DROP)

Reply:
8.0 has atomic DDL but only for InnoDB

Reply:
This behaviour is consistent with behaviour where binlog_ddl_query_log_events is 'off'.It's just that instead of built_query , we are using thd->query()

Reply:
If I consider the general use case, I feel a new option that enables "comments" and drops functionality is not right.

Reply:
For multi table drop statements , I am raising error before DDL start .
```
if (table_count > 1) {
      my_error(ER_DROP_MULTI_TABLE, MYF(0), "binlog_ddl_query_log_events");
      return true;
    }
```

Reply:
Customer feels option 2  "Add new option and with option enabled, Disallow multi table DROP (throw error to client)" is better.

Reply:
Right, proceeding with 2..